kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt

```diff
@@ -0,0 +1,400 @@
+                sectionButton.text = gardenSection.title
+                sectionButton.setOnClickListener {
+                    if (gardenSection.title == sectionButton.text) {
+                        GardenDescriptionFragment(gardenSection, this).show(
+                            supportFragmentManager,
+                            "GardenDescriptionFragment"
+                        )
+                    }
+                }
+            } else {
+                poiList.add(gardenSection)
+                poiListAdapter.notifyDataSetChanged()
+                Log.e("ADDED: ", gardenSection.title)
+                activityMainBinding.listAvailable.visibility = View.GONE
+            }
+        } else {
+            Toast.makeText(this, "Garden Section is null", Toast.LENGTH_SHORT).show()
+            Log.e("NullPointerException: ", "GardenSection is null")
+        }
+
+    }
+
+    /**
+     * Function to retrieve the image and generate a URI using [gardenSection]. The image URI is
+     * returned using a [callback] interface
+     */
+    fun retrieveImage(gardenSection: GardenSection?, callback: (uriResult: String?) -> Unit) {
+
+        val attachments = gardenSection?.attachments
+        // get the first (and only) attachment for the feature, which is an image
+        val imageAttachment = attachments?.first()
+        // fetch the attachment's data
+        val attachmentDataFuture = imageAttachment?.fetchDataAsync()
+        // listen for fetch data to complete
+        attachmentDataFuture?.addDoneListener {
+            // get the attachments data as an input stream
+            val attachmentInputStream = attachmentDataFuture.get()
+            // save the input stream to the device and get a reference to it's URI
+            val attachmentImageURI =
+                saveToInternalStorage(gardenSection.title.trim(), attachmentInputStream)
+            callback.invoke(attachmentImageURI)
+        }
+    }
+
+    /**
+     * Caches the InputStream to a BitmapImage in internal storage.
+     * Returns a String of the image file's absolutePath
+     */
+    private fun saveToInternalStorage(name: String, imageInputStream: InputStream): String {
+
+        val imagePath = File(cacheDir, name.filter { !it.isWhitespace() })
+        var fileOutputStream: FileOutputStream? = null
+        try {
+            fileOutputStream = FileOutputStream(imagePath)
+            // Use the compress method on the BitMap object to write image to the OutputStream
+            val bitmapImage = BitmapFactory.decodeStream(imageInputStream)
+            bitmapImage.compress(Bitmap.CompressFormat.PNG, 100, fileOutputStream)
+        } catch (e: Exception) {
+            Log.e("Exception: ", e.message.toString())
+            e.printStackTrace()
+        } finally {
+            try {
+                fileOutputStream?.close()
+            } catch (e: IOException) {
+                Log.e("IOException: ", e.message.toString())
```

Comment:
Check some of the other samples for formatting for Log statements. We usually use a TAG with the sample's name as the first argument